android/app/src/main/java/org/fossasia/openevent/viewmodels/SpeakersListFragmentViewModel.java

```diff
@@ -27,8 +27,10 @@ public SpeakersListFragmentViewModel() {
 
     public LiveData<List<Speaker>> getSpeakers(int sortType, String searchText) {
         setSearchText(searchText);
-        if (sortType != speakersListSortType) {
-            filterableRealmLiveData = RealmDataRepository.asFilterableLiveData(realmRepo.getSpeakers(sortOrderSpeaker()));
+        if (sortType != speakersListSortType || filterableRealmLiveData == null) {
+            if(filterableRealmLiveData == null) {
+                filterableRealmLiveData = RealmDataRepository.asFilterableLiveData(realmRepo.getSpeakers(sortOrderSpeaker()));
+            }
```

Comment:
```Java
if(filterableRealmLiveData == null) {
    filterableRealmLiveData = RealmDataRepository.asFilterableLiveData(realmRepo.getSpeakers(sortOrderSpeaker()));
}
````

This line can be moved out of the if statement I think?

Reply:
And handled above?

Reply:
@iamareebjamal Yes.I suggested him to add it but I missed the fact that filterableRealmLiveData will also change when the sortOrder is different.